org.adempiere.ui/src/org/compiere/print/DrillReportCtl.java

```diff
@@ -222,30 +222,35 @@ public void initDrillTableMap() {
 				+ "FROM AD_Column c "
 				+ " INNER JOIN AD_Column used ON (c.ColumnName=used.ColumnName)"
 				+ " INNER JOIN AD_Table t ON (used.AD_Table_ID=t.AD_Table_ID AND t.IsView='N' AND t.AD_Table_ID <> c.AD_Table_ID AND t.IsShowInDrillOptions='Y')"
+				+ " INNER JOIN AD_Tab tab ON (t.AD_Table_ID = tab.AD_Table_ID AND tab.isActive = 'Y') "
+				+ " INNER JOIN AD_Window_Access w ON (tab.AD_Window_ID = w.AD_Window_ID AND w.isActive = 'Y') "
 				+ " INNER JOIN AD_Column cKey ON (t.AD_Table_ID=cKey.AD_Table_ID AND cKey.IsKey='Y')"
 				+ " INNER JOIN AD_Element e ON (cKey.ColumnName=e.ColumnName) "
-				+ "WHERE c.AD_Table_ID=? AND c.IsKey='Y' "
-				+ "ORDER BY 3 ";
+				+ "WHERE c.AD_Table_ID=? AND w.AD_Role_ID = ? AND c.IsKey='Y' "
+				+ "GROUP BY 1, 2, 3, 4 ORDER BY 3 ";
```

Comment:
group by not needed here.

Reply:
@hengsin 

**This is an example of executing the same query without group by clause:**
![image](https://github.com/idempiere/idempiere/assets/71560285/543981a4-6f2b-4e19-8312-a6f5f15b8b77)

I used Group  By clause to avoid tables to be printed more than one time when the same table is referenced from different windows. Do you see other option to avoid it?

Reply:
To remove duplicate, should use distinct instead of group by. Group by usually for things like sum, average, etc.

Reply:
@hengsin 
Done!